examples/dtls-sock/dtls-client.c

```diff
@@ -122,26 +123,30 @@ static int client_send(char *addr_str, char *data, size_t datalen)
         return -1;
     }
 
-    res = sock_dtls_session_create(&dtls_sock, &remote, &session);
-    if (res < 0) {
+    res = sock_dtls_session_init(&dtls_sock, &remote, &session);
+    if (res <= 0) {
+        return res;
+    }
+
+    res = sock_dtls_recv(&dtls_sock, &session, buf, sizeof(buf),
+                         SOCK_NO_TIMEOUT);
+    if (res != 0) {
         printf("Error creating session: %d\n", (int)res);
         sock_dtls_close(&dtls_sock);
         sock_udp_close(&udp_sock);
         return -1;
     }
 
-    if (sock_dtls_send(&dtls_sock, &session, data, datalen) < 0) {
+    if (sock_dtls_send(&dtls_sock, &session, data, datalen, 0) < 0) {
         puts("Error sending data");
     }
     else {
         printf("Sent DTLS message\n");
 
         uint8_t rcv[512];
-        if (sock_dtls_recv(&dtls_sock, &session, rcv, sizeof(rcv), SOCK_NO_TIMEOUT) < 0) {
-            printf("Error receiving DTLS message\n");
-        }
-        else {
-            printf("Received DTLS message\n");
+        if ((res = sock_dtls_recv(&dtls_sock, &session, rcv, sizeof(rcv),
+                                    SOCK_NO_TIMEOUT)) > 0) {
+            printf("Received %d bytes\n", (int)res);
```

Comment:
Why not print the message?

Reply:
```suggestion
            printf("Received %d bytes: \"%.*s\"\n", (int)res, (int)res, (char *)rcv);
```

Reply:
Added this to the client and server and squashed.

Reply:
I'm not sure this is a good idea for the server, mostly application-wise:

1. The server is able to handle any received data, including non-printable data, while the client only expects what it sends (a printable string), since the server is an echo server.
2. The server is an echo server, so why does it need to print the data it echos?

I'd say, keep printing just the number of bytes for the server.